pages/implementation-status.md

```diff
@@ -0,0 +1,202 @@
+
+## 2023
+
+Our main objectives for 2023 will be to
+- develop the missing features of the language;
+- investigate optimal implementation strategies implement an efficient compiler;
+- implement two-way interoperability with C++; and
+- develop an inclusive community.
+
+### Implement guaranteed optimizations
+
+Val is designed in such a way that it can communicate runtime costs very transparently.
+As a result, the language can offer several guaranteed optimization.
+For example, Val guarantees that closures do not require heap allocation unless the type of their environment is erased.
+We plan on implementing these guaranteed optimizations over Q1 and Q2.
+
+### Investigate implementation strategies for concurrency
+
+Once we will have finalized the design of Val's concurrency model, we will be able to investigate the optimal implementation strategies.
+
+We are leaning toward a [stackful coroutines](https://dl.acm.org/doi/10.1145/1462166.1462167), mostly to avoid monadic asynchronous programming (a.k.a. [polychromatic functions](https://journal.stuffwithstuff.com/2015/02/01/what-color-is-your-function/)), where asynchronous operations can only be called from asynchronous contexts.
+Nonetheless, discussions with experts have revealed that such a strategy may induce unfortunate runtime costs and that stackless coroutines may not necessarily force a monadic asynchronous style upon the language.
+
+One solution might be to adopt a sackful model as a default strategy and surface stack information in the type system to let the compiler opt for stackless code generation whenever possible.
+
+We on plan having adopted and implemented a strategy by the end of Q3.
+
+### Implement two-way interoperability with C++
+
+The objective of the second phase of C++ interoperability is to support seamless interaction of Val and C++ code in the same project.
+To that end, we will interface with clang to let Val compiler understand C++ and generate appropriate wrappers.
+
+The main challenge will be to map C++ types to Val, as the former supports features that have no obvious equivalent in the latter (e.g., throwing destructors).
+We also expect our investigation and implementation to raise a plethora of new open questions.
+As such, we do not plan on delivering interoperability before Q4.
+
+### Design and implement missing language features
+
+Some of the features on our wishlist are out of scope for 2022, as we believe they are not essential to assess the validity of the language's design.
+2023 will allow us to revisit these features with a working experimental implementation.
+
+Notable features in our wishlist include:
+- exception handling;
+- polymorphic effects;
+- variadic generic value and type parameters; and
+- resumable functions (a.k.a. [generators](https://en.wikipedia.org/wiki/Generator_(computer_programming))).
+
+### Develop an inclusive community
```

Comment:
This deserves more prominence, and maybe a place on the timeline, especially if we decide to try to compete with Carbon.  They've put a lot of attention on that, or at least talked about it a lot, and they've included lots of people.